### PR TITLE
DiffWave Vocoder Added

### PR DESCRIPTION
**Pull request by VocodexElysium**
The training code and the pre-trained model are now available in Amphion. Here are the links for the pre-trained model:
[HuggingFace](https://huggingface.co/amphion/diffwave)

### CODE REVIEW

File: egs/vocoder/diffusion/README.md
Comment (RMSnow) [inline egs/vocoder/diffusion/README.md:1]: Well written! You can add the pretrained model's badge of huggingface. You can borrow the writing of [SVC recipe](https://github.com/open-mmlab/Amphion/tree/main/egs/svc/MultipleContentsSVC).

File: models/vocoders/diffusion/diffwave/diffwave.py [hunk 1]
```diff
@@ -26,6 +26,11 @@ def silu(x):
     return x * torch.sigmoid(x)
 
 
+# This model code is adopted from DiffWave/model.py under the Apache License
```
Comment (RMSnow) [inline models/vocoders/diffusion/diffwave/diffwave.py:29]: Change line29-31 into the header comments.

File: pretrained/README.md
Comment (RMSnow) [inline pretrained/README.md:1]: Replace the old one drive uploading way with the huggingface's style.

**Review by RMSnow — CHANGES_REQUESTED** [review]
Well done! Could you attach some synthesized samples of the pretrained DiffWave? Github has supported to upload files in PR. Just upload here.

File: models/vocoders/diffusion/diffwave/diffwave.py (hunk 1 above)
Comment (VocodexElysium) [reply models/vocoders/diffusion/diffwave/diffwave.py]: Finished

File: pretrained/README.md
Comment (VocodexElysium) [reply pretrained/README.md:1]: Finished

**Comment by VocodexElysium** [comment]
> Well done! Could you attach some synthesized samples of the pretrained DiffWave? Github has supported to upload files in PR. Just upload here.

Sure, here are the audio samples.

https://github.com/open-mmlab/Amphion/assets/109473203/4526ba5c-e4fc-44cb-a26d-bb5da6121ca5


https://github.com/open-mmlab/Amphion/assets/109473203/9f2f305a-f4f3-45ed-bec4-fd1d1e94d3c2


https://github.com/open-mmlab/Amphion/assets/109473203/acb42fd3-0d3d-4339-ad54-bb03705f9d84


https://github.com/open-mmlab/Amphion/assets/109473203/b51a0acd-af2e-4b62-bdbc-ef45cd617e66


https://github.com/open-mmlab/Amphion/assets/109473203/14a9c8e6-17d4-4142-a741-c77c6bb97c69

File: egs/vocoder/diffusion/README.md
Comment (VocodexElysium) [reply egs/vocoder/diffusion/README.md:1]: Badges added

**Comment by VocodexElysium** [comment]
# DiffWave Audio Samples

## Singing Voice

### Sample1

#### Gt

https://github.com/open-mmlab/Amphion/assets/109473203/d2ad7108-da44-4784-be91-45359aca2c0a

#### Pred

https://github.com/open-mmlab/Amphion/assets/109473203/545b622e-293b-4463-a414-d54208c1210e

### Sample2 

#### Gt

https://github.com/open-mmlab/Amphion/assets/109473203/56f311c9-fa56-4bbe-a3fc-e6bd62138683

#### Pred

https://github.com/open-mmlab/Amphion/assets/109473203/9765a8c7-8600-4e83-ac77-bf43b7d793a5

## Speech

### Sample1

#### Gt

https://github.com/open-mmlab/Amphion/assets/109473203/d38a0201-e80f-453a-88bc-30170ea46f4d

#### Pred

https://github.com/open-mmlab/Amphion/assets/109473203/1893e8f4-c893-4b64-9a38-91571fdb201f

### Sample2

#### Gt

https://github.com/open-mmlab/Amphion/assets/109473203/5e2b851a-5c56-481f-a29c-f40fb399f84a

#### Pred

https://github.com/open-mmlab/Amphion/assets/109473203/c04aafef-e263-4056-8dcf-100c9f27409a

**Review by RMSnow — APPROVED** [review]
Great!

**Pull request merged**